File: exemples/trustline-setup/setup_trustline.py
from xrpl.clients import JsonRpcClient
from xrpl.wallet import Wallet
from xrpl.models.transactions import TrustSet
from xrpl.transaction import send_reliable_submission

# Connect to the XRPL Testnet
client = JsonRpcClient("https://s.altnet.rippletest.net:51234/")

# Replace with your wallet's secret and issuer details
WALLET_SECRET = "s████████████████████████████"
TOKEN_ISSUER = "r████████████████████████████"
TOKEN_CURRENCY = "XGC"

# Initialize wallet
wallet = Wallet(seed=WALLET_SECRET, sequence=0)

# Create a TrustSet transaction
trust_set = TrustSet(
    account=wallet.classic_address,
    limit_amount={
        "currency": TOKEN_CURRENCY,
        "issuer": TOKEN_ISSUER,
        "value": "1000000"  # Adjust the limit as needed
    }
)

# Submit the transaction
response = send_reliable_submission(trust_set, client, wallet)

# Print the result
print("Transaction result:", response.result["meta"]["TransactionResult"])

File: exemples/trustline-setup/setup_trustline.js
const xrpl = require("xrpl");

async function main() {
    // Connect to the XRPL Testnet
    const client = new xrpl.Client("wss://s.altnet.rippletest.net:51233");
    await client.connect();

    // Replace with your wallet's secret and issuer details
    const WALLET_SECRET = "s████████████████████████████";
    const TOKEN_ISSUER = "r████████████████████████████";
    const TOKEN_CURRENCY = "XGC";

    // Initialize wallet
    const wallet = xrpl.Wallet.fromSeed(WALLET_SECRET);

    // Prepare TrustSet transaction
    const trustSetTx = {
        TransactionType: "TrustSet",
        Account: wallet.address,
        LimitAmount: {
            currency: TOKEN_CURRENCY,
            issuer: TOKEN_ISSUER,
            value: "1000000" // Adjust the limit as needed
        }
    };

    // Submit the transaction
    const response = await client.submitAndWait(trustSetTx, { wallet });
    console.log("Transaction result:", response.result.meta.TransactionResult);

    client.disconnect();
}

main();
